src/grammar.pegjs

```diff
@@ -633,20 +633,22 @@ regexp = '/' value:$(([^/\n] / '\\/')+)? closer:'/'? {
   };
 }
 
-quotedTerminal = '`' value:$(([^`\n] / ('\\`'))+)? closer:'`' {
+quotedTerminal = '`' value:$(([^`\n] / ('\\`'))+)? closer:'`' quantifier:('?')? {
   if (value === null || closer === null) {
     error('Malformed quoted terminal');
   }
   return {
     type: 'Terminal',
-    value: value
+    value: value,
+    isOptional: quantifier === '?'
   };
 }
 
-terminal = value:$(([^ \n"/`] [^ \n"\`,\]\}]*)) {
+terminal = value:$(([^ ?\n"/`] [^ ?\n"\`,\]\}]*)) quantifier:('?')? {
```

Comment:
I don't think it's necessary to change here as well. `?` as part of a terminal should be allowed. I think it is okay to require quoting the terminal to apply a `?` to remove ambiguity.